### ISSUE TITLE
tsconfig `references` not supported by Turbopack

### ISSUE THREAD

**Issue by sibbng** [vercel/next.js#68614]
### Link to the code that reproduces this issue

https://github.com/sibbng/next-paths-repro

### To Reproduce

1. Run dev script
2. Visit site
3. Check terminal

### Current vs. Expected behavior

When using the `references` option in `tsconfig.json` these references are not resolved, meaning that any option passed in them does not work.

The implementation for webpack uses Typescript directly to load the config.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
  Available memory (MB): 8021
  Available CPU cores: 4
Binaries:
  Node: 22.4.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-canary.106 // Latest available version is detected (15.0.0-canary.106).
  eslint-config-next: N/A
  react: 19.0.0-rc-06d0b89e-20240801
  react-dom: 19.0.0-rc-06d0b89e-20240801
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by timneutkens** [comment]
I've updated the description after looking at the reproduction, reproduction is still valid but the root cause is different 👍
For now I'd recommend to just move the config into the `tsconfig.json` file which will fully unblock you.

**Comment by seanparmelee** [comment]
@timneutkens the [`next.config.mjs`](https://github.com/sibbng/next-paths-repro/blob/f22fbd63aa2baf1dae6ba8bcedca86ff67d2dfb5/next.config.mjs#L6-L8) file has: 
```js
typescript: {
  tsconfigPath: './tsconfig.app.json',
}
```

Is this option irrelevant when using Turbopack and instead the root `tsconfig.json` is used?

**Comment by timneutkens** [comment]
@seanparmelee looks like that option is indeed not supported with Turbopack yet. Can you open a separate issue?

**Comment by seanparmelee** [comment]
@timneutkens FYI: https://github.com/vercel/next.js/issues/72346

**Comment by timneutkens** [comment]
`references` are now supported. Verified on next@latest: https://codesandbox.io/p/github/sibbng/next-paths-repro/csb-fmd5pd/draft/young-night
